File: src/net/dinkla/raytracer/gui/swing/ImageCanvas.java
package net.dinkla.raytracer.gui.swing;

/*
 * Copyright (c) 2012 by Jörn Dinkla, www.dinkla.com, All rights reserved.
 */

import java.awt.*;

class ImageCanvas extends Canvas {
    Image img;

    ImageCanvas(final Image img) {
        this.img = img;
    }

    public void paint(final Graphics g) {
        g.drawImage(img, 0, 0, this);
    }

    public Dimension getPreferredSize() {
        return new Dimension(img.getWidth(this), img.getHeight(this));
    }

    public Dimension getMinimumSize() {
        return getPreferredSize();
    }

}

File: src/net/dinkla/raytracer/gui/swing/ImageCanvas.kt
package net.dinkla.raytracer.gui.swing

import java.awt.*

internal class ImageCanvas(var img: Image) : Canvas() {

    override fun paint(g: Graphics) {
        g.drawImage(img, 0, 0, this)
    }

    override fun getPreferredSize(): Dimension {
        return Dimension(img.getWidth(this), img.getHeight(this))
    }

    override fun getMinimumSize(): Dimension {
        return preferredSize
    }

}
